File: Stack/Valid Parentheses/ValidParentheses.cs
public class Solution {
    public bool IsValid(string s) {
        Stack<char> st = new Stack<char>();
        foreach (char i in s)
        {
            if (st.Count == 0 && (i == ')' || i == ']' || i == '}'))
            {
                return false;
            }
            else if (i == '(' || i == '[' || i == '{')
            {
                st.Push(i);
            }
            else if (st.Count > 0 && st.Peek() == '[' && i == ']')
            {
                st.Pop();
            }
            else if (st.Count > 0 && st.Peek() == '(' && i == ')')
            {
                st.Pop();
            }
            else if (st.Count > 0 && st.Peek() == '{' && i == '}')
            {
                st.Pop();
            }
            else
            {
                return false;
            }
        }
        if (st.Count == 0)
            return true;
        return false;
    }
}

File: Stack/Valid Parentheses/ValidParentheses.go
func isValid(s string) bool {
	st := arraystack.New()
	for _, i := range s {
		if st.Empty() && (i == ')' || i == ']' || i == '}') {
			return false
		} else if i == '(' || i == '[' || i == '{' {
			st.Push(i)
		} else if value, ok := st.Peek(); !ok {
			return false
		} else if value == '[' && i == ']' {
			st.Pop()
		} else if value == '(' && i == ')' {
			st.Pop()
		} else if value == '{' && i == '}' {
			st.Pop()
		} else {
			return false
		}
	}
	if st.Empty() {
		return true
	}
	return false
}